features/logout/impl/src/test/kotlin/io/element/android/features/logout/impl/LogoutViewTest.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.logout.impl
+
+import androidx.compose.ui.test.hasContentDescription
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.performClick
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.element.android.libraries.architecture.AsyncAction
+import io.element.android.tests.testutils.EnsureCalledOnce
+import io.element.android.tests.testutils.EnsureCalledOnceWithParam
+import io.element.android.tests.testutils.EnsureNeverCalled
+import io.element.android.tests.testutils.EnsureNeverCalledWithParam
+import io.element.android.tests.testutils.EventsRecorder
+import io.element.android.tests.testutils.clickOn
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class LogoutViewTest {
+
+    @get:Rule val rule = createComposeRule()
+
+    @Test
+    fun `clicking on logout sends a LogoutEvents`() {
+        val eventsRecorder = EventsRecorder<LogoutEvents>()
+        rule.setContent {
+            LogoutView(
+                aLogoutState(
+                    eventSink = eventsRecorder
+                ),
+                onChangeRecoveryKeyClicked = EnsureNeverCalled(),
+                onBackClicked = EnsureNeverCalled(),
+                onSuccessLogout = EnsureNeverCalledWithParam(),
+            )
+        }
+        rule.clickOn("Sign out")
+        eventsRecorder.assertSingle(LogoutEvents.Logout(false))
+    }
+
+    @Test
+    fun `clicking on back invoke back callback`() {
+        val eventsRecorder = EventsRecorder<LogoutEvents>(expectEvents = false)
+        val callback = EnsureCalledOnce()
+        rule.setContent {
+            LogoutView(
+                aLogoutState(
+                    eventSink = eventsRecorder
+                ),
+                onChangeRecoveryKeyClicked = EnsureNeverCalled(),
+                onBackClicked = callback,
+                onSuccessLogout = EnsureNeverCalledWithParam(),
+            )
+        }
+        rule.onNode(hasContentDescription("Back")).performClick()
+        callback.assertSuccess()
```

Comment:
I would like to find a way to avoid having to write this line. Because if the developer forget it, the test will always pass.

Reply:
Maybe create a wrapper, like `roboTest { ... }` that internally always calls this?

Reply:
I have added `fun run` in 2c667a0dc0a00dff33ec19b7ee97310316376e5f, maybe not the best solution, but works for now I guess. But I'll think about it again.

Reply:
Improved a bit in 2a6a3ded5e95f0462cfa17cedd25dcd6deeea3b9